src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -112,19 +152,60 @@ internal static class Unix
                 string exe_dir,
                 hostfxr_get_available_sdks_result_fn result);
 
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = UTF8)]
+            internal delegate void hostfxr_get_dotnet_environment_info_result_fn(
+                IntPtr info,
+                IntPtr result_context);
+
+            [DllImport("hostfxr", CharSet = UTF8, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+            internal static extern int hostfxr_get_dotnet_environment_info(
+                string dotnet_root,
+                IntPtr reserved,
+                hostfxr_get_dotnet_environment_info_result_fn result,
+                IntPtr result_context);
+
             [DllImport("libc", CharSet = UTF8, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
             private static extern IntPtr realpath(string path, IntPtr buffer);
 
             [DllImport("libc", ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
             private static extern void free(IntPtr ptr);
 
-            internal static string realpath(string path)
+            public static string realpath(string path)
             {
                 var ptr = realpath(path, IntPtr.Zero);
                 var result = PtrToStringUTF8(ptr);
                 free(ptr);
                 return result;
             }
+
+            [StructLayout(LayoutKind.Sequential, CharSet = UTF8)]
+            internal struct hostfxr_dotnet_environment_info
+            {
+                public int size;
+                public string hostfxr_version;
+                public string hostfxr_commit_hash;
+                public int sdk_count;
+                public IntPtr sdks;
+                public int framework_count;
+                public IntPtr frameworks;
+            }
+
+            [StructLayout(LayoutKind.Sequential, CharSet = UTF8)]
+            internal struct hostfxr_dotnet_environment_framework_info
+            {
+                public int size;
+                public string name;
+                public string version;
+                public string path;
+            }
+
+            [StructLayout(LayoutKind.Sequential, CharSet = UTF8)]
```

Comment:
I am curious though why these are UTF8 on Windows. I was under the impression Charset.Unicode is required or are these not implemented using wchar to ensure compatibility with *nix?

Reply:
The strings are UTF16 on Windows and UTF8 everywhere else.